src/main/java/org/opensearch/sdk/SDKClient.java

```diff
@@ -545,6 +549,11 @@ public SDKClusterAdminClient(ClusterClient clusterClient) {
     public static class SDKIndicesClient {
 
         private final IndicesClient indicesClient;
+        private RequestOptions options = RequestOptions.DEFAULT;
+
+        public void setOptions(RequestOptions options) {
+            this.options = options;
+        }
```

Comment:
@dbwiddis do you think we can pull this method to outer class and let inner class use the same and not duplciate it?

Reply:
> @dbwiddis do you think we can pull this method to outer class and let inner class use the same and not duplciate it?

@owaiskazi19 No, because it may change per-call.

From what I understand of the security framework (And @peternied or @scrawfor99 can correct me if I'm wrong) an incoming REST request from a user will get authenticated and then have a short-lived access token associated with that request.   This token will be forwarded in the `ExtensionRestRequest` object (we currently have a placeholder for it).

At that point we will need to do something with it that makes it accessible in the RestRequest being processed by Extensions. We haven't figured this part out yet.  It may be that we just add a header when passing the rest request, consisting of the desired header(s) to send back. 

Because of the need to close these clients when we're done with them, in my ideal world we do all our REST work inside a try-with-resources block, so in a RestHandler class we would:
```java
try (SDKRestClient restClient = sdkClient.initializeRestClient()) {
    RequestOptions options = RequestOptions.Builder().addHeader("Authorization", "Bearer TOKEN_HERE").build();
    restClient.setOptions(options);
    restClient.indices().search(foo);
} // client is auto-closed here
```
Or better, change the return value of setOptions to be `this` and do a fluent call
```java
restClient.setOptions(options).indices().search(foo);
```

There's still the question of how to get either this client (with its options) or the options themselves, from the Rest Handler to the Transport Action, but I presume we can add a field in the request being sent via `client.execute()`.

However, all of this assumes setting the options in a temporary SDKRestClient object that will go away and we'll re-initialize again with a different set of options the next time around.   If we put it up in the higher level SDKClient, that's a singleton and we'd share the same token/options/whatever everywhere.  That won't work well with multithreading/async calls.

Reply:
Makes sense! Tokens are short-lived. So it's fine to keep setter different for the clients.